slack/web/classes/elements.py

```diff
@@ -1,478 +1,1042 @@
-import random
+import copy
+import logging
 import re
-import string
-from abc import ABCMeta, abstractmethod
+import warnings
+from abc import ABCMeta
 from typing import List, Optional, Set, Union
 
-from . import EnumValidator, JsonObject, JsonValidator, extract_json
-from .objects import ButtonStyles, ConfirmObject, Option, OptionGroup, PlainTextObject
+from . import EnumValidator, JsonObject, JsonValidator, show_unknown_key_warning
+from .objects import (
+    ButtonStyles,
+    ConfirmObject,
+    Option,
+    OptionGroup,
+    TextObject,
+    PlainTextObject,
+    MarkdownTextObject,
+)
+
+
+# -------------------------------------------------
+# Base Classes
+# -------------------------------------------------
 
 
 class BlockElement(JsonObject, metaclass=ABCMeta):
-    def __init__(self, *, subtype: str):
-        self.subtype = subtype
+    """Block Elements are things that exists inside of your Blocks.
+    https://api.slack.com/reference/block-kit/block-elements
+    """
+
+    attributes = {"type"}
+
+    def _subtype_warning(self):
```

Comment:
We can safely use `type` as class/instance fields while it's not available for local variables.